Clean up README for clarity (#279)
TBD54566975/ssi-service: README.md

# ssi-service

> HTTP-API driven web service that wraps our [ssi-sdk](https://github.com/TBD54566975/ssi-sdk).

Reviewer: Add a bit more?
HTTP-API driven web service that wraps our ssi-sdk and allows interacting with the standards around Verifiable Credentials, Decentralized Identifiers and all things Self Sovereign Identity!

Author: The sentence below in Introduction section has some of that, let me know if you still want to add this.

Reviewer 2: +1 on the the elevator pitch right at the top here. Suggestion pulling from both of y'all:

"A web service that exposes the `ssi-sdk` as an HTTP API. Support operations for Verifiable Credentials, Decentralized Identifiers and things Self Sovereign Identity!"

"HTTP-API driven web service" sounds semantically odd to me. :)

Author: And where andrew would you like me to add this, here or at the top?

## Introduction
The Self Sovereign Identity Service (SSIS) facilitates all things relating to [DIDs](https://www.w3.org/TR/did-core/)
and [Verifiable Credentials](https://www.w3.org/TR/vc-data-model) - in a box! The service is a part of a larger
Decentralized Web Platform architecture which you can learn more about in our

> [collaboration repo](https://github.com/TBD54566975/collaboration).

Reviewer 3: can we change this link to point to the dev website?

Author: the ssi-service page on the dev site is just this README file...https://developer.tbd.website/projects/ssi-service/README

so i'd say either docs home page: https://developer.tbd.website/docs/intro or ssi-service docs page: https://developer.tbd.website/docs/apis/ssi-service

thoughts?

Reviewer 3: docs home page

Reviewer 2: +1 to repointing this somewhere on the Dev Site. 

I think we should do two things - 

1) Point this to the Web5 projects page (https://developer.tbd.website/projects/web5), as that's what it's referencing. 
2) We rev that Projects page to be more clear, have better calls to action, more concretely outline the components which make up Web5.

Offshoring to docs home doesn't make sense to me as the link target here.

Author: So the only thing i don't love about the projects page is it's a lot of words in a developers face and the first one DIDs links to w3c so you've already lost me in a way

Reviewer 2: Yes, the Projects page absolutely needs a refresh. https://github.com/TBD54566975/developer.tbd.website/issues/326

My recommendation is to have ssi-service point to Web5 Projects page, and we make sure that one does what it needs to do on the Developer Site.

Author: So for now are you saying point to projects page or after we redo the projects page we will point to it?

Reviewer 2: Totally your call so you can close out on this issue without blocking on more work.

Medium-term I'm seeing how important it is we evolve the Project/Component landing pages on the Dev Site. Your links here deserve to point somewhere great.

## Core Functionality
- Interacting with the standards around Verifiable Credentials

> - Credential Revocations

Reviewer: and now
- Credential Suspension

> - Interacting with the standards around Verifiable Credentials
> - Credential Revocations
> - Requesting Credentials
> - Exchanging Credentials
> - Data Schemas (for credentials and other verifiable data)
> - Messaging using Decentralized Web Nodes (DWNs)
> - Usage of Decentralized Identifiers

Reviewer 3: ```suggestion
- Create and manage Decentralized Identifiers
- Create and manage Verifiable Credentials
- Interacting with the standards around Verifiable Credentials such as
  - Credential Revocations
  - Applying for Credentials
  - Exchanging Credentials
  - Data Schemas (for credentials and other verifiable data)
```

### Verifiable Interactions <br />
- Create
- Sign
- Issue
- Curate
- Request
- Revoke
- Exchange
- Validate

> - Verify

Reviewer 3: I would remove this

Author: Got it ok!

## Use Cases (more to come!)
### Business: Issuing Verifiable Credentials <br />
[Follow Tutorial](https://developer.tbd.website/docs/tutorials/issue-verifiable-credential-manually)

Steps to issue an Employment Status Credential:
1. Spin up and host the SSI-Service
2. Add the ability for your employees to click 'apply for a credential' on your internal EMS (should we show a front end button code example)
3. [Create an Issuer DID](blob/main/integration/common.go#L36) for your business
4. [Create a Schema](blob/main/integration/common.go#L66)
5. [Create a Credential Manifest](blob/main/integration/common.go#L154)
6. [Submit a Credential Application](blob/main/integration/common.go#L173)
7. Allow the ability for your employees to [revoke their credential](where?)

> 'ADD IMAGE OF THIS FLOW THAT NEAL MADE'

Reviewer: should we add this image before shipping it? 

Let me know if I need to spice it up / if you need me to find it again!

Author: Are you saying to wait till changes in this repo are shipped? or before just shipping the readme changes?

Author: @nitro-neal ?

Author: Where can folks revoke, what function should I point ppl to? @decentralgabe @nitro-neal @mistermoe ?

`Allow the ability for your employees to [revoke their credential](where?)`

Author: @decentralgabe @nitro-neal checking again

## Configuration

Managed via:
[TOML](https://toml.io/en/) [file](https://github.com/TBD54566975/ssi-service/blob/main/config/config.toml)

There are sets of configuration values for the server (e.g. which port to listen on), the services (e.g. which database to use),
and each service. Each service may define specific configuration, such as which DID methods are enabled for the DID
service.

> ### Steps for SSI-Service to consume it's configuration:

Reviewer 2: Nit: "its configuration"

mage cbt
```

> 

Reviewer 2: Why remove the pointer to CI? Would be great for folks to have a good pointer to where jobs are run and the build history incl logs. More appropriate likely in `CONTRIBUTING` rather than the `README`, I agree; how about verifying these links are right and moving this section there?

Same can be said then for the "Building" and "Deployment" sections too - `CONTRIBUTING` for devs of this project, `README` for consumers.

Author: I don't know what happened here, i didnt mean to delete this or what this is

Author: this wasn't deleted in the code I still have ...

```A utility is provided to run _clean, build, and test_ in sequence with:

```
mage cbt
```

Reviewer 2: Looks either resolved now *or* IDK what I could have possibly been talking about; there are no diffs on this line in the PR now.

}
```

## Continuous Integration

CI is managed via [GitHub Actions](https://github.com/TBD54566975/ssi-service/actions). Actions are triggered to run
for each Pull Request, and on merge to `main`. You can run CI locally using a tool
like [act](https://github.com/nektos/act).

## HTTP Endpoints
You can find all HTTP endpoints by checking out the swagger docs at: `http://localhost:8002/docs`

Note: Your port by differ, the range of the ports for swagger are between `8002` and `8080`.

## What's Supported?
- [x] [DID Management](https://www.w3.org/TR/did-core/)
  - [x] [did:key](https://w3c-ccg.github.io/did-method-key/)

>   - [ ] [did:web](https://w3c-ccg.github.io/did-method-web/)

Reviewer: did web is supported now

- [ ] Creating and managing Trust documents using [Trust Establishment](https://identity.foundation/trust-establishment/)

> - [ ] [DID Well Known Configuration](https://identity.foundation/.well-known/resources/did-configuration/) documents

Reviewer 3: this can be a check now!

Reviewer 3: lines 145,147, and 149 can all be checked

Reviewer 4: @bobbilee19 able to update that ^

- [ ] Creating and managing Trust documents using [Trust Establishment](https://identity.foundation/trust-establishment/)
- [ ] [DID Well Known Configuration](https://identity.foundation/.well-known/resources/did-configuration/) documents

## Vision, Features, and Development

The vision for the project is laid out in [this document](doc/VISION.md).

The project follows a proposal-based improvement format called [SIPs, outlined here.](sip/README.md).

Please [open a discussion](https://forums.tbd.website/c/self-sovereign-identity/16)
or [issue](https://github.com/TBD54566975/ssi-service/issues) if you are interested in helping shape the future of the project.

> 

Reviewer 2: And Discord "ssi" channel is: https://discord.com/channels/937858703112155166/969272692891086868
